conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureContexts.java

```diff
@@ -60,6 +60,14 @@ private static final class ConjureServerRequestContext implements RequestContext
             this.requestArgHandler = requestArgHandler;
         }
 
+        @Override
+        public String requestTarget() {
+            String requestUri = exchange.getRequestURI();
+            String queryString = exchange.getQueryString();
+
+            return queryString.isEmpty() ? requestUri : requestUri + "?" + queryString;
```

Comment:
If the request target has `?` but no query string, then the returned value will not include the `?`.

For example, given a request target `foo/bar?`, this implementation will return `foo/bar`.

Undertow [doesn't differentiate](https://github.com/undertow-io/undertow/blob/a55aeac10ece93b172f9175b06145d804f4b97c1/core/src/main/java/io/undertow/server/HttpServerExchange.java#L207-L210) between an unset query string and an empty query string, so there's nothing we can do about this.

In practice this should be acceptable.